### DISCUSSION TITLE
POST: /self-service/login API, the response time is unstable

### DISCUSSION THREAD

**Discussion by 704403314** [ory/kratos#3539] [Q&A]
POST: /self-service/login 
The response time is unstable. Sometimes it is only 1 second, and sometimes it exceeds 5 seconds. The front-end code controls the request timeout. If it exceeds 5 seconds, the request will be canceled.

Does anyone know the reason for the long response time of the interface?

oryd/kratos image:  v0.10.1
The language used in the backend is go

![image](https://github.com/ory/kratos/assets/17963725/eba6b02f-713b-4e6d-b3db-9a00a4e02c14)

**Comment by vinckr** [comment]
Hello @704403314 

Hmm, I don't know enough about your setup to give you a confident answer.

However, response times can be influenced by a variety of factors including server load, network conditions, and the complexity of the operations being performed. It's also important to ensure that your implementation follows the recommended practices for handling login flows as described in the [Ory Kratos User Login Documentation](https://www.ory.sh/docs/kratos/self-service/flows/user-login).
For more specific troubleshooting, you might find it helpful to refer to the [Ory Kratos General Troubleshooting Guide](https://www.ory.sh/docs/kratos/debug/troubleshooting).
